js/apps/admin-ui/src/clients/add/CapabilityConfig.tsx

```diff
@@ -234,6 +234,37 @@ export const CapabilityConfig = ({
                   )}
                 />
               </GridItem>
+              {isFeatureEnabled(Feature.StandardTokenExchangeV2) && (
+                <GridItem lg={8} sm={6}>
+                  <Controller
+                    name={convertAttributeNameToForm<
+                      Required<ClientRepresentation["attributes"]>
+                    >("attributes.standard.token.exchange.enabled")}
+                    defaultValue={false}
+                    control={control}
+                    render={({ field }) => (
+                      <InputGroup>
+                        <InputGroupItem>
+                          <Checkbox
+                            data-testid="standard-token-exchange-enabled"
+                            label={t("standardTokenExchangeEnabled")}
+                            id="kc-standard-token-exchange-enabled"
+                            name="standard-token-exchange-enabled"
+                            isChecked={field.value.toString() === "true"}
+                            onChange={field.onChange}
+                          />
```

Comment:
As the public clients are not allowed to do V2 standard token exchange (issue https://github.com/keycloak/keycloak/issues/37111) maybe we can disable this if authentication is not enabled. WDYT? (It would be copying more or less the `serviceAccountsEnabled` but only using `serviceAccountsEnabled`.)

Reply:
Yes, it makes sense to me. Thanks for raising this @rmartinc !